### ISSUE TITLE
Have an option to fail the build if subscriber is not public?

### ISSUE THREAD

**Issue by mbrubin56** [greenrobot/EventBus#253]
Thanks so much for improving an already great library! I see that if a class with an @Subscribe event handler is not public, the annotation processor prints out a message and then defaults to using reflection. It would be great to have an option to cause the compile to fail if the annotation processor encounters a non-public subscriber.

**Comment by AlokBansal8** [comment]
Yes that would be a great way to ensure that reflection is not used in our code.

**Comment by CoXier** [comment]
Hi @mbrubin56 ,can I ask you a question?

```java
subscription.subscriberMethod.method.invoke(subscription.subscriber, event);
```
`Method.invode` is a reflection?

**Comment by mbrubin56** [comment]
@CoXier I don't know where that line of code is from, so I can't say for sure. But it looks like a likely instance of [calling a method by reflection](https://docs.oracle.com/javase/8/docs/api/java/lang/reflect/Method.html#invoke-java.lang.Object-java.lang.Object...-). But without context as to why you're asking, it's hard for me to give you a good answer to what you're trying to find out.

**Comment by CoXier** [comment]
I just want to know `Method.invoke(...)` is implemented by reflection?
